proto/teleport/quicpeering/v1alpha/dial.proto

```diff
@@ -0,0 +1,57 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package teleport.quicpeering.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/teleport/quicpeering/v1alpha;quicpeeringv1alpha";
+
+// Sent from the client to the server in a fresh QUIC stream to dial a server
+// through a QUIC proxy peering connection. The message is sent in protobuf
+// binary format, prefixed by its length encoded as a little endian uint32.
+message DialRequest {
+  string server_id = 1;
+  string conn_type = 2;
+
+  Addr source = 3;
+  Addr destination = 4;
```

Comment:
The source is the address of the user initiating the connection to the proxy acting as the proxy peering client, which gets propagated to the final target of the connection; the destination is intended to act as a hint for direct dialing that will never take place since this is a proxy peering connection but we're like three levels of vestigiality deep into reverse tunnels here - morally (and practically), it's the `LocalAddr` of the reified `net.Conn` finally handled by the agent at the opposite end of the connection than the user.

Reply:
Explained in f804827f05